paasta_tools/cli/cmds/wait_for_deployment.py

```diff
@@ -137,9 +138,16 @@ def validate_git_sha(git_sha, git_url, deploy_group, service):
     Raise exception when the provided git_sha is not the latest
     marked for deployment in 'deploy_group' for 'service'.
```

Comment:
It takes 2 second to validate this.
Looks like a reasonable tradeoff to me to avoid long waiting for something that wasn't even marked for deployment. Take a look at PAASTA-8570 for instance.